### ISSUE TITLE
Possible issue with nulls and small buffer sizes in fs2

### ISSUE THREAD

**Issue by wjoel** [plokhotnyuk/jsoniter-scala#492]
Please see the test cases in https://github.com/wjoel/jsoniter-scala-null-issue (they all pass after increasing `chunkSize` to 300)

`jsoniter-scala` seems to have problems handling nulls correctly when reading files using `fs2.io.file.readAll` using a chunk size with issmall relative to the file size.

For larger files the chunk size can not be large enough without quickly running out of heap space.

Interestingly enough, the placement of the nulls seem to matter, and the parsing succeeds when nulls are placed last in the objects.

The parsing succeeds if the destination type has fields matching the source, regardless of the order of nulls in the source data and the order of the fields in the destination type.

`circe-fs2` does not have any problems in these tests.

The failing test has the following exception message:
```
expected '}' or ',', offset: 0x00000065, buf:
+----------+-------------------------------------------------+------------------+
|          |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f | 0123456789abcdef |
+----------+-------------------------------------------------+------------------+
| 00000060 |          75 6c 6c 2c 22 6e 75 6d 22 3a 31 7d 0a |    ull,"num":1}. |
| 00000070 | 7b 22 6e 61 6d 65 22 3a 22 77 69 6c 6d 61 22 2c | {"name":"wilma", |
| 00000080 | 22 65 6d 70 74 79 22 3a 6e 75 6c 6c 2c 22 6e 75 | "empty":null,"nu |
| 00000090 | 6d 22 3a                                        | m":              |
```

**Comment by plokhotnyuk** [comment]
@wjoel Thanks for reporting, Joel!

The bug affects the parsing from `java.io.InputStream` or `java.nio.DirectByteBuffer` when optional `null`, `true`, and `false` values appear on borders of chunks.

But the source of a problem was missing generative tests for skipping of JSON values :)

Please pick v2.1.11 release that fixes it.

BTW, `BufferedInputStream` is redundant [here](https://github.com/wjoel/jsoniter-scala-null-issue/blob/master/src/test/scala/JsoniterNullIssueSpec.scala#L222). The reader already does buffering when parsing from `java.io.InputStream` or `java.nio.DirectByteBuffer`. For tuning a size of the internal buffer, please, use [the `ReaderConfig.preferredBufSize` option](https://github.com/plokhotnyuk/jsoniter-scala/blob/master/jsoniter-scala-core/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/core/JsonReader.scala#L49).

**Comment by wjoel** [comment]
Awesome, thanks for the fix @plokhotnyuk! I've updated the repository with a link to this issue and a note that it's fixed in 2.1.11 and later.
